modules/gha/workflow.go

```diff
@@ -90,6 +90,7 @@ type DispatchInput struct {
 type Job struct {
 	RunsOn         []string          `json:"runs-on" yaml:"runs-on"`
 	Permissions    *JobPermissions   `json:"permissions,omitempty" yaml:"permissions,omitempty"`
+	RunIf          string            `json:"if,omitempty" yaml:"if,omitempty"`
```

Comment:
If the new focus is to map closely to the low-level GHA API, it's probably better to call this field `If` rather than `RunIf`? Better to avoid unnecessary editorializing in this new design focus IMO.

Reply:
Actually this may not be possible - I now remember hitting this error the first time when I also assumed that `If` would work:

```console
dagger --mod=.github call generate export --path=.
✔ connect 0.4s
✘ initialize 1.8s
! input: moduleSource.withContextDirectory.asModule resolve: failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to get modified source directory for go module sdk codegen: select: process "codegen --output /src --module-context-path /src/.github --module-name ci --introspection-json-path /schema.json" did not complete successfully: exit code: 1
  ✔ resolving module ref 0.2s
  ✘ installing module 1.6s
  ! input: moduleSource.withContextDirectory.asModule resolve: failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to get modified source directory for go module sdk codegen: select: process "codegen --output /src --module-context-path /src/.github --module-name ci --introspection-json-path /schema.json" did not complete successfully: exit code: 1
    ✔ ModuleSource.resolveFromCaller: ModuleSource! 0.2s
    ✘ ModuleSource.asModule: Module! 1.4s
    ! failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to get modified source directory for go module sdk codegen: select: process "codegen --output /src --module-context-path /src/.github --module-name ci --introspection-json-path /schema.json" did not complete successfully: exit code: 1

Error: input: moduleSource.withContextDirectory.asModule resolve: failed to create module: select: failed to update codegen and runtime: failed to generate code: failed to get modified source directory for go module sdk codegen: select: process "codegen --output /src --module-context-path /src/.github --module-name ci --introspection-json-path /schema.json" did not complete successfully: exit code: 1

Stdout:
generating go module: ci
Stderr:
Error: generate code: error formatting generated code: 5306:9: expected '}', found 'if' (and 10 more errors)
error: Recipe `gha` failed on line 16 with exit code 1
```

Reply:
Uh oh... I see, you hit a codegen bug! Nevermind then.